packages/adminrouter/extra/src/README.md

```diff
@@ -31,6 +31,230 @@ Use `make check-api-docs` to validate that the YAML and HTML files are up to dat
+ * `agent.conf` contains all the things that are common for all agents across
+   all flavours
+ * `master.conf` contains all the things that are common for all masters across
+   all flavours
+Each section can have either `ee` or `open` directories, but never both. `ee`
+directory is only present in EE repo, Open repo contains only `open` directories.
+Contents of these directories may be as follows:
+ * `common.conf` contains all the things that are common for given flavour, no
+   matter the server type
+ * `agent.conf` contains all the things that are common for agents in given
+   flavour
+ * `master.conf` contains all the things that are common for masters in given
+   flavour
+The order of includes is for the time being hard-coded in nginx.\*.conf files,
+but it does not matter that much as:
+ * [nginx by default orders the imports while globbing](https://serverfault.com/questions/361134/nginx-includes-config-files-not-in-order)
+ * the order of includes should not matter within a particular section. This means
+   that e.g. location blocks *MUST NOT* rely on the order in which they appear
+   in the config file and define matching rules precise enough to avoid ambiguity.
+
+All the includes are bound together by nginx.(master|agent).conf files.
+`nginx.master.conf` for `Open` master looks like this ATM:
+```
+include includes/main/common.conf;
+include includes/main/open/common.conf;
+
+http {
+    include includes/http/common.conf;
+    include includes/http/master.conf;
+    include includes/http/open/common.conf;
+    include includes/http/open/master.conf;
+
+    server {
+        server_name master.mesos leader.mesos;
+
+        include includes/server/common.conf;
+        include includes/server/master.conf;
+        include includes/server/open/master.conf;
+
+        include /opt/mesosphere/etc/adminrouter-listen-open.conf;
+        include /opt/mesosphere/etc/adminrouter-upstreams-open.conf;
+        include /opt/mesosphere/etc/adminrouter-tls.conf;
+    }
+}
+```
+and for the agent:
+```
+include includes/main/common.conf;
+include includes/main/open/common.conf;
+
+http {
+    include includes/http/common.conf;
+    include includes/http/agent.conf;
+    include includes/http/open/common.conf;
+
+    server {
+        server_name agent.mesos;
+
+        include includes/server/common.conf;
+        include includes/server/open/agent.conf;
+
+        include /opt/mesosphere/etc/adminrouter-listen-open.conf;
+        include /opt/mesosphere/etc/adminrouter-tls.conf;
+    }
+}
+```
+
+### Enforcing code reuse
+The EE repo contains only ee directories, all common/agent-common/master-common
+code resides in Open repository. This way EE repository becomes an overlay on
+top of Open. Only `nginx.(master|agent).conf` are overwritten while applying EE
+repository on top of Open during DC/OS image build. EE DC/OS image build scripts
+remove all open directories from the Open repository before applying ee repo on
```

Comment:
❓ Would be good to have a consistent way to refer to the repositories. So far in this block "ee repo" "EE repository".